Validation/EcalDigis/interface/EcalSelectiveReadoutValidation.h

```diff
@@ -7,47 +7,32 @@
  *
  */
 
-#include "FWCore/Framework/interface/Frameworkfwd.h"
-#include "FWCore/Framework/interface/EDAnalyzer.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+#include "DQMServices/Core/interface/MonitorElement.h"
 
-#include "DataFormats/Common/interface/Handle.h"
 #include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
 #include "SimDataFormats/CaloHit/interface/PCaloHit.h"
 
-
-#include "Geometry/CaloTopology/interface/EcalTrigTowerConstituentsMap.h"
-#include "DataFormats/EcalDetId/interface/EcalScDetId.h"
-#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
-
 #include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
-//#include "DataFormats/EcalDetId/interface/EEDetId.h"
-//#include "DataFormats/EcalDetId/interface/EBDetId.h"
 #include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
-#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
 #include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/EcalDetId/interface/EcalScDetId.h"
+#include "DataFormats/EcalDetId/interface/EcalTrigTowerDetId.h"
 
 #include "Validation/EcalDigis/src/CollHandle.h"
 
 #include <string>
 #include <set>
-#include <utility>
 #include <fstream>
-#include <inttypes.h>
-#include "DQMServices/Core/interface/DQMStore.h"
-#include "DQMServices/Core/interface/MonitorElement.h"
 
-class DQMStore;
-class MonitorElement;
 class EBDetId;
 class EEDetId;
-class EcalTPParameters;
 class EcalElectronicsMapping;
+class EcalTrigTowerConstituentsMap;
 
-class EcalSelectiveReadoutValidation: public edm::EDAnalyzer{
+class EcalSelectiveReadoutValidation: public DQMEDAnalyzer{
```

Comment:
Hi, @Dr15Jones 

Thank you for looking at the SelectiveReadout module. I will fix the "firstCall" code as you suggested, and create a patch for it.

cheers,
Menglei